src/Text/Layout/Table/Cell/WideString.hs

```diff
@@ -21,26 +21,44 @@ instance Cell WideString where
     visibleLength (WideString s) = realLength s
     measureAlignment p (WideString s) = measureAlignmentWide p s
     buildCell (WideString s) = buildCell s
-    buildCellView = buildCellViewLRHelper
+    buildCellView = buildCellViewHelper
       (\(WideString s) -> buildCell s)
-      (\i (WideString s) -> WideString $ dropWide True i s)
-      (\i (WideString s) -> WideString . reverse . dropWide False i $ reverse s)
+      (\i (WideString s) -> buildCell $ trimLeft i s)
+      (\i (WideString s) -> buildCell $ trimRight i s)
+      (\l r (WideString s) ->
+          -- First, try to trim from the right
+          let CellView a _ rightAdj = trimRight r s in
+          -- If there is extra padding, trim less on the left
+          buildCell $ trimLeft (l - rightAdj) a
+      )
 
 -- | Drop characters from the left side of a 'String' until at least the
 -- provided width has been removed.
 --
 -- The provided `Bool` determines whether to continue dropping zero-width
 -- characters after the requested width has been dropped.
-dropWide :: Bool -> Int -> String -> String
-dropWide _ _ [] = []
+dropWide :: Bool -> Int -> String -> CellView String
+dropWide _ _ [] = pure ""
 dropWide gobbleZeroWidth i l@(x : xs)
     | gobbleZeroWidth && i == 0 && charLen == 0 = dropWide gobbleZeroWidth i xs
-    | i <= 0       = l
+    | i <= 0       = pure l
     | charLen <= i = dropWide gobbleZeroWidth (i - charLen) xs
-    | otherwise    = replicate (charLen - i) ' ' ++ dropWide gobbleZeroWidth 0 xs
+    | otherwise    = adjustCell (charLen - i) 0 =<< dropWide gobbleZeroWidth 0 xs
   where
     charLen = charWidth x
 
+-- | Drop characters from the left side of a 'String', recording how much padding
+-- is needed to have dropped the requested width.
+trimLeft :: Int -> String -> CellView String
+trimLeft = dropWide True
+
+-- | Drop characters from the right side of a 'String', recording how much padding
+-- is needed to have dropped the requested width.
```

Comment:
This sentence sounds weird to me.  Maybe it is just me.

Reply:
Altered the wording.